src/Analyzer.Utilities/DiagnosticHelpers.cs

```diff
@@ -98,5 +99,29 @@ public static string GetMemberName(ISymbol symbol)
             // For other language constructs
             return symbol.Name;
         }
+
+        public static string GetMeaningfulParentName(SyntaxNode current, SemanticModel model)
```

Comment:
Could you please add a comment there and move it to Desktop analyzer's helper class?